controllers/nestedcluster_controller.go

```diff
@@ -48,10 +59,51 @@ type NestedClusterReconciler struct {
 }
 
 // SetupWithManager sets up the controller with the Manager.
-func (r *NestedClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
+func (r *NestedClusterReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
+	clusterToInfraFn := util.ClusterToInfrastructureMapFunc(infrav1.GroupVersion.WithKind("NestedCluster"))
+	log := ctrl.LoggerFrom(ctx)
+
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&infrav1.NestedCluster{}).
-		Owns(&controlplanev1.NestedControlPlane{}).
+		WithOptions(options).
+		For(&infrav1.NestedCluster{},
+			builder.WithPredicates(
+				predicate.Funcs{
+					// Avoid reconciling if the event triggering the reconciliation is related to incremental status updates
+					UpdateFunc: func(e event.UpdateEvent) bool {
+						oldCluster := e.ObjectOld.(*infrav1.NestedCluster).DeepCopy()
+						newCluster := e.ObjectNew.(*infrav1.NestedCluster).DeepCopy()
+						oldCluster.Status = infrav1.NestedClusterStatus{}
+						newCluster.Status = infrav1.NestedClusterStatus{}
+						oldCluster.ObjectMeta.ResourceVersion = ""
+						newCluster.ObjectMeta.ResourceVersion = ""
+						return !reflect.DeepEqual(oldCluster, newCluster)
```

Comment:
I think it is OK for now since there is only one field (`Ready`) in the `NestedCluster.Status`. @christopherhein what's your opinion?

Reply:
I think this is fine, cause we don't really need to rereconcile after we set `Ready`.